tools/build_script_generator/cmake/module.lb

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017-2018, Fabian Greif
+# Copyright (c) 2017-2018, Niklas Hauser
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+import os
+
+
+def init(module):
+    module.name = "build.cmake"
+    module.description = """CMake Build Script Generator
+
+Generates a CMakeLists.txt file to build the modm library.
+"""
+
+
+def prepare(module, options):
+    _, default_project_name = os.path.split(os.getcwd())
+
+    module.add_option(
+        StringOption(name="build.path", default=".",
+                     description="Path to the build folder"))
+    module.add_option(
+        StringOption(name="project.name", default=default_project_name.replace('.', '_'),
+                     description="Project name (required)"))
+
+    if options[":target"].identifier["platform"] == "stm32":
+        return True
+    return False
+
+
+def build(env):
+    pass
+
+
+def post_build(env, buildlog):
+    target = env["modm:target"]
+    platform = target.identifier["platform"]
+    family = target.identifier["family"]
+
+    # Extract all source code files
+    files_to_build = []
+    outpath = env.outpath(".")
+    for operations in buildlog:
+        filename = os.path.relpath(operations.filename_out, outpath)
+        _, extension = os.path.splitext(filename)
+
+        if extension in [".c", ".cpp", ".cc", ".sx", ".S"]:
+            files_to_build.append(os.path.normpath(filename))
+    files_to_build.sort()
+
+    core_driver = target.get_driver("core")
+    core = core_driver["type"]
+    if core.startswith("cortex-m4"):
+        core = "cortex-m4"
+    elif core.startswith("cortex-m7"):
+        core = "cortex-m7"
+
+    memories = []
+    if "memory" in core_driver:
+        memories.extend([
+            {
+                k:(int(v) if v.isdigit() else (int(v, 16) if v.startswith("0x") else v))
+                for k, v in memory.items()
+            }
+            for memory in core_driver["memory"]
+        ])
+    #----------------------------------------------------------------------------
+    # When out-of-source modm source tree build (for example, if modm included
+    # as a submodule into the project), is performed, configuration file name
+    # is prepended with wrong directory name - 'modm' instead of 'board'
+    #----------------------------------------------------------------------------
+    config_file = ''
+    if buildlog.metadata["openocd.configfile"]:
+        config_file = buildlog.metadata["openocd.configfile"][0]
+        head,tail = os.path.split(config_file)
+
+        if head == 'modm' :
+            config_file = 'board/'+tail
```

Comment:
I forgot to finish porting the support for custom OpenOCD support.
I've done this [now in this commit](https://github.com/modm-io/modm/commit/06b3c81f52e69ca373da9ac56510c4ccad5c8b28).

The idea is to prefix custom config scripts with `modm/` to signify to the user that these config scripts are NOT portable across machines! So the prefix is actually intentional, but the files were never copied. I've taken your suggestion and put the board config files into `modm/board/*.cfg` to mirror the directory structure built into OpenOCD.

Note that you need to add the generated `modm/openocd/` folder to your OpenOCD searchpath.
This is done with the `-s` option.

Reply:
Please note that you need to use my patched version of lbuild to use this commit, see https://github.com/dergraaf/library-builder/pull/26.
```sh
pip install --upgrade git+https://github.com/dergraaf/library-builder@refs/pull/26/head
```

cc @dergraaf

Reply:
This part has been merged and I'm working on getting lbuild into PyPI. That should make it easier to install.